File: solutions/typescript/acronym/1/acronym.ts
export function parse(phrase: string): string {
    const words: string [] = [];

    const parts: string[] = phrase.split(new RegExp("-| "));
    parts.forEach(part => {
        //split camel case
        const split = part.match(/([A-Z]+(?=[A-Z][a-z])|[A-Z]?[a-z]+)/g) || [];
        if(split.length > 0){
            words.push(...split);
        }
        else {
            words.push(part);
        }
    });

    let acronym: string = "";
    console.warn(words)
    words.forEach((word: string) => {
        acronym += word[0];
    });

    return acronym.toUpperCase();
}

File: solutions/java/acronym/1/src/main/java/Acronym.java
import java.util.*;

class Acronym {
    private final String iPhrase;

    Acronym(String phrase) {
        this.iPhrase = phrase;
    }

    String get() {
        String[][] words = Arrays.stream(this.iPhrase.split(" |-"))
                .map(w -> w.toUpperCase().replaceAll("[^0-9A-Z]", "").split(""))
                .toArray(String[][]::new);

        String acronym = "";
        for(String[] w: words){
            acronym += w[0];
        }

        return acronym;
    }
}